server.js

```diff
@@ -1,23 +1,95 @@
 const express = require("express");
 const cors = require("cors");
-
+const { request, response } = require("express");
 const app = express();
 
+
+const bodyParser = require("body-parser")
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}))
+
 app.use(cors());
+app.use(express.json());
+
+app.use((request, response, next) => {
+  request.body.timeSent = new Date();
+  next();
+});
 
 const welcomeMessage = {
   id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
+  from: "Neill",
+  text:"hi hi"
 };
-
 //This array is our "data store".
-//We will start with one message in the array.
+//oneWe will start with  message in the array.
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+
+app.get("/messages",(request,response)=>{
+  console.log(request.body.timeSent);
+  response.status(200).json({messages})
+})
+app.get("/messages/search", (request, response) => {
+  const searchText = request.query.text;
+  let filteredMessages = messages;
+  if (searchText) {
+    filteredMessages = messages.filter((message) =>
+      message.text.includes(searchText)
+    );
+  }
+  response.json({ filteredMessages });
+});
+app.get("messages/latest", (request, response) => {
+  const searchText = request.query.text;
+  let recentMessages = messages.slice(-10);
+  if (searchText) {
```

Comment:
do you really need this block, or it's just a copy-pasty error ?

Reply:
If you mean the if block,I wanted to check if the value of query is exist